### PR TITLE
Bug 1278396 - Improve performance by adding/removing webviews instead of layering them

### PR DESCRIPTION
**Pull request by tylerlacroix**

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -2255,7 +2255,18 @@ extension BrowserViewController: WKNavigationDelegate {
             // forward/backward. Strange, but LayoutChanged fixes that.
             UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, nil)
         } else {
-            screenshotHelper.takeDelayedScreenshot(tab)
+            // To Screenshot a tab that is hidden we must add the webView,
+            // then wait enough time for the webview to render.
+            if let webView =  tab.webView {
+                view.insertSubview(webView, atIndex: 0)
+                let time = dispatch_time(DISPATCH_TIME_NOW, Int64(500 * NSEC_PER_MSEC))
```
Comment (sleroux) [inline Client/Frontend/Browser/BrowserViewController.swift]: There's a method in `ScreenshotHelper` that does just this called `takePendingScreenshot`.

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -1939,6 +1934,7 @@ extension BrowserViewController: TabManagerDelegate {
             // we therefore have to hide all the scrollViews that we are no actually interesting in interacting with
             // to ensure that scrollsToTop actually works
             wv.scrollView.hidden = true
```
Comment (fluffyemily) [inline Client/Frontend/Browser/BrowserViewController.swift]: would be worth testing, but if we are removing the `webView` we no longer need to hide the scroll view in order to enable `scrollToTop`, so this line can be deleted

Comment (fluffyemily) [inline Client/Frontend/Browser/BrowserViewController.swift]: We could also then remove lines 1947 - 1949 below where we unhide the webview if it is hidden.

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 1 above)
Comment (tylerlacroix) [inline Client/Frontend/Browser/BrowserViewController.swift]: I tried using takePendingScreenshot but it didn't work. The only issue with takePendingScreenshot is that it calls the takeScreenshot which then calls "drawViewHierarchyInRect", and that method can only take a screenshot of views that are currently being rendered on the screen.  So if we add a new tab in the background, it's not being rendered so the screenshot will be white until we click on the tab (and the view is rendered).  What this does is just adds the views just long enough to render and take a screen shot, then remove it.

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 2 above)
Comment (tylerlacroix) [inline Client/Frontend/Browser/BrowserViewController.swift]: Okay! I'll test removing those lines! :)

**Pull request merged**